lib/puppetlabs_spec_helper/rake_tasks.rb

```diff
@@ -75,217 +76,246 @@
-def vagrant_ssh(set, node = nil)
-  vagrant_yml_dir = File.join '.vagrant', 'beaker_vagrant_files', "#{set}.yml"
-  vagrant_file = File.join vagrant_yml_dir, 'Vagrantfile'
-  unless File.file? vagrant_file
-    puts "There is no Vagrantfile at: '#{vagrant_file}'. Perhaps, the node is not created or is destroyed."
-    exit 1
-  end
-  Dir.chdir(vagrant_yml_dir) do
-    command = 'vagrant ssh'
-    command += " #{node}" if node
-    # Vagrant is not distributed as a normal gem
-    # and we should protect it from the current Ruby environment
-    env = {
-        'RUBYLIB' => nil,
-        'GEM_PATH' => nil,
-        'BUNDLE_BIN_PATH' => nil,
-    }
-    system env, command
+  # get the array of Beaker set names
+  # @return [Array<String>]
+  def beaker_node_sets
+    return @beaker_nodes if @beaker_nodes
+    @beaker_nodes = Dir['spec/acceptance/nodesets/*.yml'].sort.map do |node_set|
+      node_set.slice!('.yml')
+      File.basename(node_set)
+    end
   end
-end
-
-def auto_symlink
-  { File.basename(Dir.pwd).split('-').last => '#{source_dir}' }
-end
 
-def fixtures(category)
-  if File.exists?('.fixtures.yml')
-    fixtures_yaml = '.fixtures.yml'
-  elsif File.exists?('.fixtures.yaml')
-    fixtures_yaml = '.fixtures.yaml'
-  else
-    fixtures_yaml = ''
+  # Use "vagrant ssh" to login to the given node in the node set
+  # @param set [String] The name of the node set (yml file)
+  # @param node [String] The name of the node in the set. For multi-node sets.
+  def vagrant_ssh(set, node = nil)
+    vagrant_yml_dir = File.join '.vagrant', 'beaker_vagrant_files', "#{set}.yml"
+    vagrant_file = File.join vagrant_yml_dir, 'Vagrantfile'
+    unless File.file? vagrant_file
+      puts "There is no Vagrantfile at: '#{vagrant_file}'. Perhaps, the node is not created or is destroyed."
+      exit 1
+    end
+    Dir.chdir(vagrant_yml_dir) do
+      command = 'vagrant ssh'
+      command += " #{node}" if node
+      # Vagrant is not distributed as a normal gem
+      # and we should protect it from the current Ruby environment
+      env = {
+          'RUBYLIB' => nil,
+          'GEM_PATH' => nil,
+          'BUNDLE_BIN_PATH' => nil,
+      }
+      system env, command
+    end
   end
 
-  begin
-    fixtures = (YAML.load_file(ENV['FIXTURES_YML'] || fixtures_yaml) || { fixtures: {} })
-  rescue Errno::ENOENT
-    fixtures = {}
-  rescue Psych::SyntaxError => e
-    abort("Found malformed YAML in #{fixtures_yaml} on line #{e.line} column #{e.column}: #{e.problem}")
+  def auto_symlink
+    { File.basename(Dir.pwd).split('-').last => '#{source_dir}' }
   end
 
-  if fixtures.include? 'defaults'
-    fixture_defaults = fixtures['defaults']
-  else
-    fixture_defaults = {}
-  end
+  def fixtures(category)
+    if ENV['FIXTURES_YML']
+      fixtures_yaml = ENV['FIXTURES_YML']
+    elsif File.exists?('.fixtures.yml')
+      fixtures_yaml = '.fixtures.yml'
+    elsif File.exists?('.fixtures.yaml')
+      fixtures_yaml = '.fixtures.yaml'
```

Comment:
it would be nice if we could move to .yaml, but that will probably never happen.